crates/derive/src/combine_api.rs

```diff
@@ -133,6 +144,10 @@ impl cgo::Service for API {
                 Ok(())
             }
             (Code::Drain, Some(mut state)) => {
+                tracing::debug!(
```

Comment:
I think this would be more useful if it measured memory now, and then logged debug with the delta after the operation completes.

nit: formatting.